### PR TITLE
[tcgc] spread behavior change

### PR DESCRIPTION
**Pull request by tadelesh**
use original model for body parameter in `SdkServiceOperation` if it is from a simple spread.

simple spread means spread model without any `@header`, `@query` or `@path` property and the spread model is the only source of http body.

examples, 

`op test(...Model): void;`

`op test(@header h: string, @query q: string, ...Model): void;`

these two operation will have `Model` as body parameter.

```
op test(...A): void;
model A {
  @query
  q: string;
  prop: string;
}
```
this operation will have an anonymous model with only `prop` property and generated name `TestRequest` as body parameter.

resolve: https://github.com/Azure/typespec-azure/issues/1132

cc: @ArcturusZhang @joheredi

### CODE REVIEW

**Review by iscai-msft — APPROVED** [review]

File: packages/typespec-client-generator-core/src/internal-utils.ts [hunk 1]
```diff
@@ -546,3 +553,41 @@ export function isXmlContentType(contentType: string): boolean {
...
+
+/**
+ * If body is from spread, then it does not directly from a model property.
+ * @param httpBody
+ * @param parameters
+ * @returns
+ */
+export function isHttpBodySpread(
+  httpBody: HttpOperationBody | HttpOperationMultipartBody,
+  parameters: Model
```
[the remark this answers is not in the document]
Comment (tadelesh) [reply packages/typespec-client-generator-core/src/internal-utils.ts:565]: good catch, when i change the function logic, forgot to change the signature.

**Comment by tadelesh** [comment]
@ArcturusZhang  @joheredi please review and check if it could satisfy your language's requirement, thanks.

**Comment by ArcturusZhang** [comment]
Do we have a dev version so that we could see if this change could meet our requirement?
The code here LGTM

**Comment by tadelesh** [comment]
> Do we have a dev version so that we could see if this change could meet our requirement? The code here LGTM

before this pr merge, you could use the ci build artifact. after merge, you could use the dev version.

**Comment by tadelesh** [comment]
@ArcturusZhang have you tried this in .net emitter? if all behavior is good, could you approve this pr?

File: packages/typespec-client-generator-core/src/internal-utils.ts [hunk 2]
```diff
@@ -546,3 +553,38 @@ export function isXmlContentType(contentType: string): boolean {
...
+ * @param type
+ * @returns
+ */
+export function getHttpBodySpreadModel(type: Model): Model {
+  if (type.sourceModels.length === 1 && type.sourceModels[0].usage === "spread") {
+    const innerModel = type.sourceModels[0].model;
+    // for case: `op test(...Model):void;`
+    if (innerModel.name !== "" && innerModel.properties.size === type.properties.size) {
+      return innerModel;
+    }
```
Comment (weidongxu-microsoft) [inline packages/typespec-client-generator-core/src/internal-utils.ts:578]: Did we consider the case that the `Model` here include `@path` `@query` etc.?

Comment (tadelesh) [reply packages/typespec-client-generator-core/src/internal-utils.ts:578]: nope, see pr description for details.

File: packages/typespec-client-generator-core/test/decorators.test.ts
[the remark this answers is not in the document]
Comment (tadelesh) [reply packages/typespec-client-generator-core/test/decorators.test.ts:1]: added more tests

File: packages/typespec-client-generator-core/src/internal-utils.ts (hunk 2 above)
Comment (weidongxu-microsoft) [reply packages/typespec-client-generator-core/src/internal-utils.ts:578]: Wondering which line exclude this scenario. `innerModel.properties.size === type.properties.size`?

If `Model` contains `@query`, would the size be different?

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**